client/devices/client.go

```diff
@@ -57,81 +59,109 @@ type Client struct {
 	url            string
 	devicesListURL string
 	client         *http.Client
+	output         io.Writer
 }
 
 func NewClient(url string, skipVerify bool) *Client {
 	return &Client{
 		url:            url,
 		devicesListURL: client.JoinURL(url, devicesListURL),
 		client:         client.NewHttpClient(skipVerify),
+		output:         os.Stdout,
```

Comment:
is there a reason we are introducing this? do we do it anywhere else?

Reply:
I needed a way to redirect the output for the unit tests. I'd like to structure things differently, but I didn't want to do more extensive refactoring upon consideration of the diff that needs reviewing.